spec/jobs/batch_create_job_spec.rb

```diff
@@ -65,5 +65,14 @@
         expect(log.reload.status).to eq 'failure'
       end
     end
+
+    context "when multiple resource types are passed" do
+      let(:resource_types) { { upload1.id.to_s => ['Image', 'Text'], upload2.id.to_s => ['Image', 'Text'] } }
+      it "sets them all on the record" do
+        subject
+        expect(GenericWork.last.reload.resource_type).to include 'Image'
```

Comment:
Good catch. I don't think I can do it that way since the order won't be guaranteed. But I can grab them once instead of twice for sure.